docs/multi-stage-query/concepts.md

```diff
@@ -102,6 +104,19 @@ issues](./known-issues.md#select) page.
 
 For more information about the syntax, see [REPLACE](./reference.md#replace).
 
+When deciding whether to use REPLACE or INSERT, keep in mind that REPLACE can perform dimension-based segment pruning but INSERT cannot. To use dimension-based segment pruning, your query needs to meet the following requirements:
```

Comment:
This part and the above change on line 81/82 needs to be rephrased.

To clarify, INSERT or REPLACE queries (better called statements for less ambiguity in this case?) cannot and need not perform dimension-based pruning (* unless they are doing a sub-SELECT on some other datasource).
It is the queries (i.e. SELECT queries of any kind) that we run on the _segments_ generated with the above INSERT or REPLACE statements that could potentially benefit from such pruning.

As a suggestion:

```suggestion
When deciding whether to use REPLACE or INSERT, keep in mind that segments generated with REPLACE can be pruned with dimension-based pruning but those generated with INSERT cannot. Please refer to [Clustering](#clustering) for the prerequisites of dimension-based pruning.:
```

Reply:
Linking to it is probably a better solution than straight repeating. Will make this change